packages/core/src/components/Menu/MenuItem/MenuItem.tsx

```diff
@@ -56,6 +56,14 @@ export interface MenuItemProps extends VibeComponentProps {
   splitMenuItem?: boolean;
   "aria-label"?: AriaAttributes["aria-label"];
   submenuPosition?: SubmenuPosition;
+  /**
+   * Enables the observation of content resize for the menu item's submenu.
+   * When set to `true`, a ResizeObserver is attached to the popper content,
+   * automatically triggering repositioning when the size of the submenu's content changes.
+   *
+   * This is useful for when submenu's content may grow or shrink without a re-render being triggered.
+   */
+  observeSubMenuContentResize?: boolean;
```

Comment:
Are the implementation details really matter here? I think we can simplify it and just say that it will respond to resizing of the content
Also worth renaming prop name IMO, maybe `autoAdjustSubMenuOnResize`?

Reply:
Sounds good